### PR TITLE
add solutions in Python

### PR DESCRIPTION
**Pull request by mafzaldev**
CHANGE LOG:
- added medium question_5.py
- added hard question_3.py

### CODE REVIEW

File: Hard/question_2.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+"""
```
Comment (saadali18) [inline Hard/question_2.py:1]: **Choose different question from hard section**
Python doesn't have the "friend" concept explicitly, it provides a different mechanism for controlling access to class members based on naming conventions and conventions

Like, 
**Public members:** Members that are part of the class's public section and are intended to be accessed by users of the class are typically named without any leading underscores. For example, self.my_variable is a public member.

**Protected members:** Members intended for internal use within the class or its subclasses are typically named with a single leading underscore, like _my_variable. 

**Private members:** Python doesn't have true private members like C++ does. Instead, members that are intended to be kept as private and not accessed from outside the class are named with double leading underscores, such as __my_variable.

File: Medium/question_5.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        self.__age = age
+
+    def __display(self):
+        print("This is private method.")
+
+
+student = Student("Bill", 25)
+print(
+    student.__school_name
+)  # AttributeError: 'Student' object has no attribute '__school_name'
```
Comment (saadali18) [inline Medium/question_5.py:72]: Make it a single liner

File: Medium/question_5.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    __school_name = "XYZ School"
+
+    def __init__(self, name, age):
+        self.__name = name
+        self.__age = age
+
+    def __display(self):
+        print("This is private method.")
+
+
```
Comment (saadali18) [inline Medium/question_5.py:68]: Add all the print statements and class initializations under 
`if __name__ == "__main__":`
Read more about it: https://stackoverflow.com/questions/419163/what-does-if-name-main-do

File: Medium/question_5.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+student = Student("Bill", 25)
+print(
+    student.__school_name
+)  # AttributeError: 'Student' object has no attribute '__school_name'
+
+print(student.__name)  # AttributeError: 'Student' object has no attribute '__name'
+
+print(
+    student.__display()
+)  # AttributeError: 'Student' object has no attribute '__display'
```
Comment (saadali18) [inline Medium/question_5.py:78]: Make it single liner

**Review by saadali18 — CHANGES_REQUESTED** [review]

File: Medium/question_5.py (hunk 4 above)
Comment (mafzaldev) [reply Medium/question_5.py]: I was using `Black` formatter for Python, which was formatting in this style. Fixed now!

File: Medium/question_5.py (hunk 2 above)
Comment (mafzaldev) [reply Medium/question_5.py]: I was using `Black` formatter for Python, which was formatting in this style. Fixed now!

**Review by saadali18 — APPROVED** [review]

**Pull request merged**